crates/core/src/partitions.rs

```diff
@@ -166,32 +164,31 @@ impl PartitionRoutingRefresher {
                 Some(cmd) = self.receiver.recv() => {
                     match cmd {
                         Command::SyncRoutingInformation => {
-                            self.spawn_sync_routing_information_task();
+                            self.spawn_sync_routing_information_task(*partition_table_watch.borrow());
                         }
                     }
                 }
-                _ = update_interval.tick() => {
+                _ = partition_table_watch.changed() => {
                     trace!("Refreshing routing information...");
-                    self.spawn_sync_routing_information_task();
+                    self.spawn_sync_routing_information_task(*partition_table_watch.borrow());
```

Comment:
If `partition_table_watch` tells us that a given partition table version is present, I think we can just call `Metadata::current(|m| m.partition_table_ref())` to get access to it. Therefore I think that spawning an additional task shouldn't be necessary.

Reply:
I totally missed that. In my head I thought we still have to wait for the version to sync not that the metadata is already up to date. Thank you for pointing this out